mitmproxy/addons/export.py

```diff
@@ -106,6 +107,69 @@ def httpie_command(f: flow.Flow) -> str:
     return cmd
 
 
+def python_requests_command(f: flow.Flow) -> str:
+    # Copy as Python Requests
+    request = cleanup_request(f)
+    request = pop_headers(request)
+
+    def escape_quotes(s: str) -> str:
+        return s.replace('"', '\\"')
+
+    code = "import requests\n\n"
+
+    code += f'url = "{request.pretty_url}"\n'
+    code += "headers = {\n"
+    cookies = ""
+    for k, v in request.headers.items(multi=True):
+        if k.lower() == "cookie":
+            cookies += f"{v};"
+        else:
+            code += f'    "{k}": "{escape_quotes(v)}",\n'
+    code += "}\n"
+
+    if cookies:
+        code += "cookies = {\n"
+        for cookie in cookies.strip().split("; "):
+            key, value = cookie.split("=", 1)
+            code += f'    "{key}": "{escape_quotes(value)}",\n'
+        code += "}\n"
+    else:
+        code += "cookies = {}\n\n"
+
+    is_json = False
+    if request.content:
+        try:
+            decoded_content = request.content.decode("utf-8")
```

Comment:
Let's streamline this:

1. If `json.loads(request.content)` works, go with that. json.loads accepts raw bytes, so we don't have to decode for that.
2. If https://docs.mitmproxy.org/stable/api/mitmproxy/http.html#Message.text does not raise, assign body_str to repr(text)
3. As a fallback, assign body_str to repr(request.content).

Technically steps 1-2 are optional, but definitely nice to have.

Reply:
Not sure how to do step 2.

Reply:
I was thinking of something like this:

```python
from pprint import pformat

try:
    body_val = json.loads(request.content)
except ValueError:
    try:
        body_val = request.text
    except ValueError:
        body_val = request.content

body_str = f"body = {pformat(body_val, indent=4, sort_dicts=False)}"
```

Reply:
And this, will create weird code format like this
```
body = {   'string': 'Hello, world!',
    'number': 42,
    'float': 3.14,
    'boolean': True,
    'nullValue': None,
    'object': {'name': 'John', 'age': 30},
    'array': [1, 2, 3, 4]}

```